### DISCUSSION TITLE
Auto Domain population or drop down selection.

### DISCUSSION THREAD

**Discussion by ffslcschools** [apache/cloudstack#9921] [Ideas]
For smaller environments where the IT department is separated by internal teams, it would be nice to have the login process either automatically select the domain associated with the account or if the user happens to be a member of multiple domains, have a drop down that highlights the allowed domains. 
I understand that this wouldn't work in an MSP scenario, but for smaller organizations that keep everything internal it would be a welcomed feature. 
I searched the discussions and did not find anything related to this so I figured I would give it a mention.

**Comment by DaanHoogland** [comment]
though it sound as great functionality, this might be a security concern @ffslcschools . It can be implemented behind a feature flag that is false by default. So only operators in a private/enterprise environment would use it.

**Reply by ffslcschools** [reply]
Yes, being able to enable/disable would be fine and I agree that it would be a feature that is off by default.

**Reply by DaanHoogland** [reply]
now for a technical challenge: we would need an API call to get the domains, while no-one is logged on to the system. That might be a blocker.

**Reply by DaanHoogland** [reply]
if your domains are rather static there could be a solution in a definition of the domains directly in the UI, btw.

**Reply by ffslcschools** [reply]
I have them setup for manual ldap account creation, so they won't change often. Point me in the direction of the definition of the domains.. I am still playing with all the scoping of roles and access to this that and the other.

**Reply by DaanHoogland** [reply]
There is a table in the DB and an API ListDomains, In the code there is a DomainVO(.java) that contains all the data. I think you are only interest in domainpath.
